tests/functional/test_config_provider.py

```diff
@@ -49,3 +53,68 @@ def test_default_configurations_resolve_correctly():
     assert client.meta.config.connect_timeout == 3.1
     assert client.meta.endpoint_url == 'https://sts.us-west-2.amazonaws.com'
     assert client.meta.config.retries['mode'] == 'standard'
+
+
+def _create_functional_test_loader():
+    test_models_dir = Path(__file__).parent / 'models' / 'test-smart-defaults'
+    loader = botocore.loaders.Loader()
+    loader.search_paths.insert(0, test_models_dir)
+    return loader
+
+
+@pytest.fixture
+def session():
+    loader = _create_functional_test_loader()
+    session = botocore.session.Session()
+    session.register_component('data_loader', loader)
+    return session
+
+
+def config_store_updated(original_config_store, updated_config_store):
+    return original_config_store is not updated_config_store
+
+
+class TestSmartDefaultsConfigStoreFactory:
```

Comment:
The name of this class is too specific to the implementation details. Let's rename it to `TestConfigurationDefaults` to emphasize we are testing end to end configuration default resolution.